### ISSUE TITLE
False positive for a method reference accessing the non-null key of a nullable map

### ISSUE THREAD

**Issue by toellrich** [uber/NullAway#1795]
## Description

NullAway reports a nullness error for a method reference that accesses the key of a map entry, even though the map's key type is non-null.

The map is declared as:

```java
Map<String, @Nullable String> map
```

Only the value type is nullable. The key type is `String` and is therefore non-null. Consequently, `entry.getKey()` must also return a non-null `String`.

The following lambda expression correctly compiles without any NullAway diagnostic:

```java
.map(entry -> entry.getKey())
```

However, replacing the lambda expression with the equivalent method reference causes NullAway to report an error.

The equivalent method reference is incorrectly rejected:

```java
.map(Entry::getKey)
```

This demonstrates that the lambda expression behavior is correct, while the method reference produces a false positive.

## Minimal Reproduction

```java
import java.util.HashMap;
import java.util.Map;
import java.util.Map.Entry;
import org.jspecify.annotations.Nullable;

class NullAwayLambdaBugReportTest {

  public static void main(String[] args) {
    Map<String, @Nullable String> map = new HashMap<>();
    map.put("foo", "bar");

    map.entrySet().stream()
        .map(Entry::getKey)
        .forEach(System.out::println);
  }
}
```

Replacing the method reference with the equivalent lambda expression makes the diagnostic disappear:

```java
map.entrySet().stream()
    .map(entry -> entry.getKey())
    .forEach(System.out::println);
```

## Expected Behavior

Both forms should compile without any NullAway diagnostic:

```java
.map(entry -> entry.getKey())
```

```java
.map(Entry::getKey)
```

The map's key type is non-null, so `Map.Entry.getKey()` should be inferred as returning a non-null `String`. The nullable value type must not affect the nullness of the key.

## Actual Behavior

NullAway reports a nullness error for the method reference:

```java
.map(Entry::getKey)
```

The equivalent lambda expression does not produce a diagnostic:

```java
.map(entry -> entry.getKey())
```

The method reference is therefore incorrectly treated as potentially returning a nullable value.

## Environment

- Java: 21
- NullAway: 0.14.1
- Error Prone: 2.50.0
- JSpecifyMode: enabled
- JSpecifyExperimental: enabled

**Comment by msridhar** [comment]
This one was fixed as part of #1804 which inadvertently got merged as part of #1807
